broadcast.go

```diff
@@ -135,35 +120,48 @@ func (broadcast *broadcast) ForEach(room string, f EachFunc) {
 }
 
 // Len gives number of connections in the room
-func (broadcast *broadcast) Len(room string) int {
-	broadcast.lock.RLock()
-	defer broadcast.lock.RUnlock()
+func (bc *broadcast) Len(room string) int {
+	bc.lock.RLock()
+	defer bc.lock.RUnlock()
 
-	return len(broadcast.rooms[room])
+	return len(bc.rooms[room])
 }
 
 // Rooms gives the list of all the rooms available for broadcast in case of
 // no connection is given, in case of a connection is given, it gives
 // list of all the rooms the connection is joined to
-func (broadcast *broadcast) Rooms(connection Conn) []string {
-	broadcast.lock.RLock()
-	defer broadcast.lock.RUnlock()
-
-	rooms := make([]string, 0)
-	if connection == nil { // create a new list of all the room names
-		// iterate through the rooms map and add the room name to the above list
-		for room := range broadcast.rooms {
+func (bc *broadcast) Rooms(connection Conn) []string {
+	bc.lock.RLock()
+	defer bc.lock.RUnlock()
+
+	if connection == nil {
+		return bc.AllRooms()
+	}
+
+	return bc.getRoomsByConn(connection)
+}
+
+// AllRooms gives list of all rooms available for broadcast
+func (bc *broadcast) AllRooms() []string {
+	bc.lock.RLock()
+	defer bc.lock.RUnlock()
+
+	rooms := make([]string, 0, len(bc.rooms))
+	for room := range bc.rooms {
+		rooms = append(rooms, room)
+	}
+
+	return rooms
+}
+
+func (bc *broadcast) getRoomsByConn(connection Conn) []string {
```

Comment:
Nice to make this public perhaps?

Reply:
It is inner method for 
```go
func (bc *broadcast) Rooms(connection Conn) []string 
```
I will suggest separate current method `Rooms` to double methods `Rooms(...)` and `AllRooms(...)`